### ISSUE TITLE
[Bug]: In-App Message's images are not visible

### ISSUE THREAD

**Issue by malcommac** [braze-inc/braze-swift-sdk#18]
### Platform

iOS

### Platform Version

16.0.3

### Braze SDK Version

5.5.1

### Xcode Version

14.0.1

### Computer Processor

Apple (M1)

### Repro Rate

100%

### Steps To Reproduce

Cannot see the In-App Messages images when sending push campaign.  
Once initialized, I've set the following parameters on the SDK:

```swift
let inAppMessageUI = BrazeInAppMessageUI()
inAppMessageUI.delegate = self
self.brazeSDK?.inAppMessagePresenter = inAppMessageUI
        
// - GIF support
GIFViewProvider.shared = .sdWebImage
```

where `.sdWebImage` is:

```swift
extension GIFViewProvider {

  /// A GIF view provider using [SDWebImage](https://github.com/SDWebImage/SDWebImage) as a
  /// rendering library.
  static let sdWebImage = Self(
    view: { SDAnimatedImageView(image: image(for: $0)) },
    updateView: { ($0 as? SDAnimatedImageView)?.image = image(for: $1) }
  )

  private static func image(for url: URL?) -> UIImage? {
    guard let url = url else { return nil }
    return url.pathExtension == "gif"
      ? SDAnimatedImage(contentsOfFile: url.path)
      : UIImage(contentsOfFile: url.path)
  }

}
```

However, I cannot see [static] images in received messages.  
Is there anything else I should do to configure the SDK properly?

On [web site I found](https://www.braze.com/docs/developer_guide/platform_integration_guides/ios/in-app_messaging/troubleshooting/#troubleshooting-in-app-message-display):

> Failed image downloads will prevent in-app messages with images from displaying. Image downloads will always fail if the SDWebImage framework is not integrated properly. Check your device logs to ensure that image downloads are not failing.

but I cannot see anything else I should do to integrate this feature.

### Expected Behavior

See the static image

### Actual Incorrect Behavior

The image is not visible.

### Verbose Logs

```shell
Nothing relevant, apparently


- date: 2022-10-13 13:25:47 +0000
- silent: true
- userInfo: {
  "ab" : {
    "c" : "dGVzdF90ZXN0X2RpPTYzNDgxMWQ5OGI4MTUxZDEyMTQ2N2I5YzIzOTEyYWQ4"
  },
  "ab_push_fetch_test_triggers_key" : true,
  "aps" : {
    "alert" : "Tap to see Test In-App Message"
  }
}
› ⚙️  [info/core]      2022-10-13 15:25:47.409 [BrazeService:213]      › In-app message ready for display:
- triggerId: (campaignId: 634811d99248f023d44aa333, raw: NjM0ODE...3Rlc3Q=)
- extras: {

}```
```

**Comment by malcommac** [comment]
<img width="687" alt="image" src="https://user-images.githubusercontent.com/235645/195609668-6d62d32f-6d03-4b49-bcf6-b681f567f73b.png">

**Comment by lowip** [comment]
Hi @malcommac,

Thanks for reporting this issue and the detailed information. Your implementation of the Braze SDK looks good and this seems to indicate a bug in either the SDK or the platform response for that campaign.

To investigate further, we will need the URL of the campaign on the Braze dashboard. Would you mind sending this information to support@braze.com, referencing this Github issue in the email as well.

Thanks,

**Comment by malcommac** [comment]
Hi @lowip, thank you for your quick reply. We have sent an email with a link to this issue.

**Comment by mathaeus** [comment]
We are experiencing the same problem.

@lowip additionally we are seeing "No network connection" for certain button clicks that can occur in the IAM (e.g. for a link to the app store)

![df122906-dd80-4937-9d97-6a317444fd0f](https://user-images.githubusercontent.com/5389635/196369708-cce0b82d-a2d7-4443-952b-889b958a1c53.png)

----

When we integrated the new SDK a few weeks ago everything seemed to be fine when we tested on a real device. Now we have the new SDK out with our users and this is impacting us quite heavily. Please fix this ASAP.

**Comment by malcommac** [comment]
Yes, it's impacting our release cycle heavily too.

**Comment by malcommac** [comment]
Any info about this error?

btw: does the SDWebImage necessary to show static images? Or it's only necessary for GIFs?

**Comment by lowip** [comment]
Hi @malcommac, @mathaeus 

We were unable to reproduce the issue (see [video](https://user-images.githubusercontent.com/503337/197062217-8b62b8f1-3cfa-4ee8-bc20-d39826e391d7.mov)). I can confirm that the campaign is correctly setup as well.

- Are you still able to reproduce the error consistently? 
  - The SDK aggressively cache images for offline use, does the issue persist after calling [`wipeData()`](https://braze-inc.github.io/braze-swift-sdk/documentation/brazekit/braze/wipedata())?
- @mathaeus Could you send a link to the campaign to [support@braze.com](mailto:support@braze.com) referencing this issue?

> additionally we are seeing "No network connection" for certain button clicks that can occur in the IAM (e.g. for a link to the app store)

This is a separate issue, we have a fix currently in review. In the meantime, you can deselect the `Open Web URL Inside App` checkbox on the campaign creation page.

> btw: does the SDWebImage necessary to show static images? Or it's only necessary for GIFs?

`SDWebImage` is only needed for GIF support, by default [`GIFViewProvider.shared`](https://braze-inc.github.io/braze-swift-sdk/documentation/brazeui/gifviewprovider-swift.struct/shared) is set to [`GIFViewProvider.nonAnimating`](https://braze-inc.github.io/braze-swift-sdk/documentation/brazeui/gifviewprovider-swift.struct/nonanimating) which simply renders all images (GIFs included in an `UIImageView`).

You can try commenting out the line `GIFViewProvider.shared = .sdWebImage` in your code to disable any kind of GIF support.

**Comment by mathaeus** [comment]
Hi @lowip, thanks for getting back to us.

> Are you still able to reproduce the error consistently?

Yes, 100%. Meaning all custom HTML IAMs fail on all devices. IAMs using the template don't have that problem AFAIK.
We have verified that the custom HTML IAMs work on older builds with the Objc-based `appboy-ios-sdk`.

> The SDK aggressively cache images for offline use, does the issue persist after calling [wipeData()](https://braze-inc.github.io/braze-swift-sdk/documentation/brazekit/braze/wipedata())?

I haven't tried that yet, but tbh that doesn't sound like it's something that can (should) be applied on a user level. I only have one testing device with me at the moment, so I'd rather try with a new campaign after there was a fix.
As I said all campaigns that we tried within the last few days are failing on all devices that we tried with, so to me it doesn't sound like a "caching" problem only.

> Could you send a link to the campaign to [support@braze.com](mailto:support@braze.com) referencing this issue?

We've already been in contact with Mike from Support for the last two days, who now should have a TestFlight build and a campaign from us.

----

> This is a separate issue, we have a fix currently in review. 

I see, thanks for the info. Is/was there a Github issue for this as well? 

> In the meantime, you can deselect the Open Web URL Inside App checkbox on the campaign creation page.

That option does not exist for custom HTML IAMs. See screenshot below.

<img width="593" alt="Screenshot 2022-10-21 at 09 51 14" src="https://user-images.githubusercontent.com/5389635/197142961-22452537-7124-4f24-834e-f30fb68c9fbb.png">

**Comment by lowip** [comment]
Hi @mathaeus, thanks for this information.

We were able to reproduce a similar issue when using the _Send Test_ button in the campaign's page on the dashboard, but not when triggering the campaign any other way like it would happen in live conditions (e.g. custom event, session start, etc.).

We have a fix in review for the _Send Test_ button issue.

Did you experience the issue only with the dashboard's _Send Test_ button or also in real-life conditions?

Regarding the "No network connection" issue, the fix has been merged and will be part of our next release.

**Comment by mathaeus** [comment]
> We were able to reproduce a similar issue when using the Send Test button in the campaign's page on the dashboard, but not when triggering the campaign any other way like it would happen in live conditions (e.g. custom event, session start, etc.).

> We have a fix in review for the Send Test button issue.

> Did you experience the issue only with the dashboard's Send Test button or also in real-life conditions?

Hi @lowip, that explains why we didn't catch this bug when we were doing our tests for the SDK upgrade, because back then we were using real triggers and not the `Send Test` functionality...
So yes, if we're using real triggers like `Custom Event` the custom HTML IAM is displayed correctly.

When can we expect a fix for the `Send Test` functionality, also in the next SDK release?

----

> Regarding the "No network connection" issue, the fix has been merged and will be part of our next release.

Cool, great to hear 👍

**Comment by malcommac** [comment]
We encountered the same error also with Send Test (I missed reporting the source of the test above).
Should the "Send Test" fix impact the SDK with another release?

**Comment by lowip** [comment]
@mathaeus, @malcommac 

> When can we expect a fix for the Send Test functionality, also in the next SDK release?

Yes, the fix is merged and we are preparing that next release which should be available later this week.

> Should the "Send Test" fix impact the SDK with another release?

The fix is on the SDK-side and will require an update.

I'll update here once that release is available.

**Comment by lowip** [comment]
@malcommac, @mathaeus 

We have released [5.6.0](https://github.com/braze-inc/braze-swift-sdk/releases/tag/5.6.0) which includes fixes for both issues mentioned in this ticket.

Thanks,

**Comment by malcommac** [comment]
just to confirm IAP message images works fine from our side
